lib/active_shipping/location.rb

```diff
@@ -73,10 +73,11 @@ def self.from(object, options = {})
 
       ATTRIBUTE_ALIASES.each do |attribute, aliases|
         aliases.detect do |sym|
-          value = if hash_access
-            object[sym]
-          elsif object.respond_to?(sym)
-            object.send(sym)
+          value = object[sym] if hash_access
+          if !value &&
+            object.respond_to?(sym) &&
+            (!hash_access || !Hash.public_instance_methods.include?(sym))
+            value = object.send(sym)
           end
```

Comment:
This is just a suggestion - Perhaps not a great one, so take with a grain of salt..